libcontainer/cgroups/fs2/psi.go

```diff
@@ -0,0 +1,78 @@
+package fs2
+
+import (
+	"bufio"
+	"fmt"
+	"os"
+	"strconv"
+	"strings"
+
+	"github.com/opencontainers/runc/libcontainer/cgroups"
+)
+
+func statPSI(dirPath string, file string, stats *cgroups.PSIStats) error {
+	f, err := cgroups.OpenFile(dirPath, file, os.O_RDONLY)
+	if err != nil {
+		return err
+	}
+	defer f.Close()
+
+	sc := bufio.NewScanner(f)
+	for sc.Scan() {
+		parts := strings.Fields(sc.Text())
+		switch parts[0] {
+		case "some":
+			data, err := parsePSIData(parts[1:])
+			if err != nil {
+				return err
+			}
+			stats.Some = data
+		case "full":
+			data, err := parsePSIData(parts[1:])
+			if err != nil {
+				return err
+			}
+			stats.Full = data
+		}
+	}
+	if err := sc.Err(); err != nil {
+		return &parseError{Path: dirPath, File: file, Err: err}
+	}
+	return nil
+}
+
+func parsePSIData(psi []string) (data cgroups.PSIData, err error) {
+	for i := 0; i < len(psi); i++ {
```

Comment:
I think it'll be more readable to use
```go
          for _, f := range psi {
```
(and use `f` instead of `psi[i]` below).